README.md

```diff
@@ -59,6 +61,31 @@ Install specific files in Dom0:
 sudo qubesctl state.apply dotfiles.copy-dom0,dotfiles.copy-sh,dotfiles.copy-vim,dotfiles.copy-x11
 ```
 
+#### Pillar
+
+With Salt, the execution of each state can be opt out via the dotfiles pillar.
+You will need a top and a sls file in your pillar_roots (default to /srv/pillar,
+/srv/pillar/base, can be configured in /etc/salt/minion.d).
+
+Exemple: /srv/pillar/qusal/dotfiles.top to apply the dotfiles sls to all targets:
+```yaml
+base:
+  '*':
+    - qusal.dotfiles
+```
+Exemple: /srv/pillar/dotfiles.sls containing the data to pass to the targets:
```

Comment:
I am thinking of formulas where we may set defaults via pillar and let the user customize it, so I still need to learn if there is something such as precedence in pillar (rendering order).

Reply:
Well, I suppose I have bad news:

Testing with two $pillar.top asking for two distinct $pillar.sls (one with qusal:dotfiles:all = true and one with qusal:dotfiles:all = false), after enabling them, I have no consistence about the values returned.

In dom0 the pillar return false, despite the latest top enabled should return true, and in appvms, it returns true.

It may be related in the fact that QubesOS implements its own mechanisms here, as I remember that I hadn't the same commands available on other OS, but I cannot say much for now about that, as my debian test server will be unavailable for a few weeks.

In my perspective, it enforces my idea that we should try to provides sane defaults directly in the states, that can be overridden later by user pillars, again, shipping and enabling pillar for the user is after all a complexity that you may want to avoid after-all.

Reply:
> Well, I suppose I have bad news:
> 
> Testing with two $pillar.top asking for two distinct $pillar.sls (one with qusal:dotfiles:all = true and one with qusal:dotfiles:all = false), after enabling them, I have no consistence about the values returned.
>
> In dom0 the pillar return false, despite the latest top enabled should return true, and in appvms, it returns true.

:(
 
> It may be related in the fact that QubesOS implements its own mechanisms here, as I remember that I hadn't the same commands available on other OS, but I cannot say much for now about that, as my debian test server will be unavailable for a few weeks.

Oh well, probably a Qubes code bug on how it is rendered for dom0.

> In my perspective, it enforces my idea that we should try to provides sane defaults directly in the states, that can be overridden later by user pillars, again, shipping and enabling pillar for the user is after all a complexity that you may want to avoid after-all.

I agree with that, I just thought it was a good idea to provide the defaults to user "documented" in a pillar file itself, which well, can still be done, but being something that is never enabled such as `pillar.default` (or a better extension that won't be enabled). What do you think?

Reply:
> I agree with that, I just thought it was a good idea to provide the defaults to user "documented" in a pillar file itself, which well, can still be done, but being something that is never enabled such as pillar.default (or a better extension that won't be enabled). What do you think?

This seems a good idea.